### PR TITLE
Support dynamic message name on Conduit.Test functions

### PR DESCRIPTION
**Pull request by youalreadydid**
Hi, my team is using this nice library and sometimes we want to test dynamic message names.

I don't know if there's a better way to support this (the code feels repetitive and couldn't keep the deprecated behaviour), so any suggestions are welcome :)

### CODE REVIEW

**Comment by blatyo** [comment]
I'm not sure I understand the use case. Name there should always be an atom because it refers to `publish :name, ...`.

**Comment by blatyo** [comment]
Oh, I see. We can go to a v1.0.0 for this.

File: lib/conduit/test.ex [hunk 1]
```diff
@@ -76,28 +76,16 @@ defmodule Conduit.Test do
 
       MyApp.Broker.publish(:message, %Conduit.Message{})
       assert_message_published(:message)
+      message_name = :message
+      assert_message_published(message_name)
 
   """
   defmacro assert_message_published(name) when is_atom(name) do
```
Comment (blatyo) [inline lib/conduit/test.ex]: Should just drop the guard here.

File: lib/conduit/test.ex [hunk 2]
```diff
@@ -76,28 +76,16 @@ defmodule Conduit.Test do
...
-
-          assert_message_published #{inspect(binding()[:name])}, #{unquote(message_code)}
-
-      Or
-
-          assert_message_published #{inspect(binding()[:name])}
-      """)
-    end
+  defmacro assert_message_published(name) do
+    quote do: assert_received({:publish, _, ^unquote(name), _, _, _})
```
Comment (blatyo) [inline lib/conduit/test.ex]: The `^` should be up to the user. So, they could do something like:

``` elixir
assert_message_received name
assert name == :foo

#or

name = :foo
assert_message_received ^name
```

File: lib/conduit/test.ex (hunk 1 above)
Comment (youalreadydid) [reply lib/conduit/test.ex]: done :)

File: lib/conduit/test.ex (hunk 2 above)
Comment (youalreadydid) [reply lib/conduit/test.ex]: good point, I've tried this yesterday but I messed up something...
done!

**Review by blatyo — APPROVED** [review]

**Pull request merged**